packages/solid-virtual/src/index.tsx

```diff
@@ -55,22 +55,19 @@ function createVirtualizerBase<
   }
 
   const virtualizer = new Proxy(instance, handler)
-  virtualizer.setOptions(resolvedOptions)
 
   onMount(() => {
     const cleanup = virtualizer._didMount()
-    virtualizer._willUpdate()
     onCleanup(cleanup)
   })
 
-  createComputed(() => {
+  createEffect(() => {
     virtualizer.setOptions(
       mergeProps(resolvedOptions, options, {
         onChange: (
           instance: Virtualizer<TScrollElement, TItemElement>,
           sync: boolean,
         ) => {
-          instance._willUpdate()
           setVirtualItems(
             reconcile(instance.getVirtualItems(), {
               key: 'index',
```

Comment:
Should also make this 'key' rather than 'index' so that it uses any custom supplied key from `getItemKey` option for the diff.